library/src/blas1/rocblas_asum.hpp

```diff
@@ -0,0 +1,45 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#pragma once
+
+#include "fetch_template.h"
+#include "handle.h"
+#include "reduction_strided_batched.h"
+#include "rocblas.h"
+
+template <class To>
+struct rocblas_fetch_asum
+{
+    template <typename Ti>
+    __forceinline__ __device__ To operator()(Ti x, ptrdiff_t)
+    {
+        return {fetch_asum(x)};
+    }
+};
+
+// allocate workspace inside this API
+template <rocblas_int NB, typename Ti, typename To>
+rocblas_status rocblas_asum_template(rocblas_handle handle,
```

Comment:
Since these templates are _exactly_ (or close to it) the same code, I still think it's a good idea to combine them into one, as Lee stated it does reduce future upkeep and I don't think it will add any complexity in this case.
At the very least we could use the `asum_strided_batched` template with a `batch_count` of 1 here. Unless we're concerned that that will degrade performance.

Reply:
All 3 call the same template kernel this style just exposes 3 internal APIs for reuse rather than 1 where you have to put in stride 0, and batch count 1 when not using those features.   Fewer parameters that match the C API makes more sense to me than having more parameters.   If we want to add another parameter variation later on do we keep adding on to a single generic one or make a new one?    Could merge later in another PR as well.

Reply:
Good point, if we want to keep the internal API clean then having extra parameters for different cases likely isn't ideal. A possible alternative if having these "internal API calls" call a generic function which can handle all three, this could improve code reuse and keep the internal API clean?